hw/syn/tools/dc/constraints.sdc

```diff
@@ -0,0 +1,160 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Generic constraints file for simple testsynthesis flow
+# This held very simple for now and needs to be refined
+
+set DEFAULT_CLOCK_UNCERTAINTY 0.5
+
+# TODO: consider splitting this into per-IP .sdc files
+if {$TOP_ENTITY == "top_earlgrey"} {
+
+puts "Applying constraints for top level"
+
+# note: this does not account for clock insertion delay and
+# there are no pads instantiated in the netlist (yet)
+
+#####################
+# main clock        #
+#####################
+set MAIN_CLK_PIN clk_i
+set MAIN_RST_PIN rst_ni
+# 125 MHz
+set MAIN_TCK  8.0
+set_ideal_network ${MAIN_CLK_PIN}
+set_ideal_network ${MAIN_RST_PIN}
+
+create_clock ${MAIN_CLK_PIN} -period ${MAIN_TCK}
+set_clock_uncertainty ${DEFAULT_CLOCK_UNCERTAINTY}  ${MAIN_CLK_PIN}
+
+set IN_DEL    5.5
+set OUT_DEL   5.5
+
+set_input_delay ${IN_DEL} [get_ports mio_in_i*]          -clock ${MAIN_CLK_PIN}
```

Comment:
I know this is first-pass, just attaching a note for future: you probably want to set a driver for inputs, especially for a block-level build, to get reasonable estimates for resistance/input risetimes.

Reply:
I actually set load and a driver to all IOs in the main synthesis script.

Reply:
maybe these load/drivers should be moved into the sdc as well...